src/branchIfEmpty.tsx

```diff
@@ -0,0 +1,30 @@
+import {ReactElement} from 'react'
+import {branch, renderNothing, returns} from 'ad-hok'
+import {isEmpty} from 'lodash'
+import {some} from 'lodash/fp'
+
+import {ensureArray} from './utils'
+
+type BranchIfEmptyType = <TProps extends {}, TPropNames extends keyof TProps>(
```

Comment:
Ah, I see why you used that example above that I commented on (because you're branching on a nested value).